tests/test_models.py

```diff
@@ -58,3 +63,39 @@ def test_calculate_support_status_system(mocker, current_date, system_start, sys
 )
 def test_get_rhel_display_name(name, major, minor, expected):
     assert _get_rhel_display_name(name, major, minor) == expected
+
+
+@pytest.mark.parametrize("count", (0, 1, 2))
+def test_system_populate_systems_from_systems_details(generate_system_detail, count):
+    system_ids = set()
+    systems_detail = set()
+
+    for _ in range(0, count):
+        system, system_id = generate_system_detail
+        system_ids.add(system_id)
+        systems_detail.add(system)
```

Comment:
This look can be eliminated by change the fixture to accept a count parameter.
```suggestion
    systems_detail = generate_system_detail(count)
```